src/thunderbots/software/ai/world/ball.h

```diff
@@ -15,20 +15,29 @@ class Ball
     explicit Ball();
 
     /**
-     * Updates the ball with new data.
+     * Updates the ball with new data, updating the current data as well as the predictive
+     * model
      *
-     * @param new_position the new position of the ball.
-     * @param new_velocity the new velocity of the ball.
+     * @param new_position the new position of the ball, defined in metres
+     * @param new_velocity the new velocity of the ball, defined in metres per second
      */
-    void update(Point &new_position, Point &new_velocity);
+    void update(Point &new_position, Vector2 &new_velocity);
```

Comment:
We should call `Vector2` `Vector` or `Vector2d`. `Vector2` makes it should like it's the second version of the `Vector` class.